### DISCUSSION TITLE
请问下大佬，为什么我在页面使用vue-cesium封装的组件，通过useVueCesium.createPromise 拿到的res为undefined, 是只能在组件内部使用useVueCesium嘛

### DISCUSSION THREAD

**Discussion by whyds000** [zouyaoji/vue-cesium#328] [Q&A]
`<template>
  <cs> </cs>
</template>

<script setup lang="ts">
import cs from '@/components/cs/cs.vue'
import { useVueCesium } from 'vue-cesium'
import { createCircle } from '../cs/lib/warning/createCirlcle'
const $vc = useVueCesium('c-container')
$vc.creatingPromise.then((res) => {
  console.log(res)

//此时res为undefined，页面能正常显示
})
</script>

<style scoped></style>
`

**Comment by zouyaoji** [comment]
是的哈，尽量像[vue-cesium-demo](https://github.com/zouyaoji/vue-cesium-demo/blob/main/src/layouts/MainLayout.vue#L56) 那样把业务组件都放vc-viewer下就没这种问题了。

如果确实不想放进去，那目前使用 useVueCesium 的地方要在 vc-viewer 的 create生命周期之后才能获取到哈。后面看看能不能优化下！

**Reply by whyds000** [reply]
好的，谢谢大佬
